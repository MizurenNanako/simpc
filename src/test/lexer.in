#include <stdio.h>
#include "linking_list.h"

void pout(llist_int *list)
{
    for (int i = 0; i < list->length; ++i)
    {
        printf("index: %d\t data: %d\n", i, query_index_list_int(list, i));
    }
    puts("");
}

int main()
{
    llist_int *list = create_list_int(0);
    for (int i = 1; i < 10; ++i)
    {
        push_back_list_int(list, i);
    }
    pout(list);
    insert_index_list_int(list, 5, 999);
    insert_index_list_int(list, 3, 777);
    pout(list);
    delete_list_nod\
e_int(list, 8);
    delete_list_node_int(list, 8);
    pout(list);
    delete_list_int(list);
    return 0;
}