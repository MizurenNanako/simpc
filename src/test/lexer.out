[line:   1, col:  19] -> typeid: 1	type: [prep]	info: "#include <stdio.h>"
[line:   2, col:   1] -> typeid: 102	symbol: "[newline]"

[line:   2, col:  26] -> typeid: 1	type: [prep]	info: "#include "linking_list.h""
[line:   3, col:   1] -> typeid: 102	symbol: "[newline]"

[line:   3, col:   5] -> typeid: 48	symbol: "void"
[line:   3, col:   9] -> typeid: 2	type: [id]	info: "pout"
[line:   3, col:  10] -> typeid: 13	symbol: "("
[line:   3, col:  19] -> typeid: 2	type: [id]	info: "llist_int"
[line:   3, col:  20] -> typeid: 79	symbol: "*"
[line:   3, col:  24] -> typeid: 2	type: [id]	info: "list"
[line:   3, col:  25] -> typeid: 14	symbol: ")"
[line:   4, col:   1] -> typeid: 102	symbol: "[newline]"

[line:   4, col:   2] -> typeid: 17	symbol: "{"
[line:   5, col:   1] -> typeid: 102	symbol: "[newline]"

[line:   5, col:   4] -> typeid: 32	symbol: "for"
[line:   5, col:   5] -> typeid: 13	symbol: "("
[line:   5, col:   8] -> typeid: 35	symbol: "int"
[line:   5, col:   9] -> typeid: 2	type: [id]	info: "i"
[line:   5, col:  10] -> typeid: 63	symbol: "="
[line:   5, col:  11] -> typeid: 3	symbol: "0"
[line:   5, col:  12] -> typeid: 101	symbol: ";"
[line:   5, col:  13] -> typeid: 2	type: [id]	info: "i"
[line:   5, col:  14] -> typeid: 92	symbol: "<"
[line:   5, col:  18] -> typeid: 2	type: [id]	info: "list"
[line:   5, col:  20] -> typeid: 97	symbol: "->"
[line:   5, col:  26] -> typeid: 2	type: [id]	info: "length"
[line:   5, col:  27] -> typeid: 101	symbol: ";"
[line:   5, col:  29] -> typeid: 74	symbol: "++"
[line:   5, col:  30] -> typeid: 2	type: [id]	info: "i"
[line:   5, col:  31] -> typeid: 14	symbol: ")"
[line:   6, col:   1] -> typeid: 102	symbol: "[newline]"

[line:   6, col:   2] -> typeid: 17	symbol: "{"
[line:   7, col:   1] -> typeid: 102	symbol: "[newline]"

[line:   7, col:   7] -> typeid: 2	type: [id]	info: "printf"
[line:   7, col:   8] -> typeid: 13	symbol: "("
[line:   7, col:  32] -> typeid: 10	type: [string]	info: ""index: %dtt data: %dnn""
[line:   7, col:  33] -> typeid: 98	symbol: ","
[line:   7, col:  34] -> typeid: 2	type: [id]	info: "i"
[line:   7, col:  35] -> typeid: 98	symbol: ","
[line:   7, col:  55] -> typeid: 2	type: [id]	info: "query_index_list_int"
[line:   7, col:  56] -> typeid: 13	symbol: "("
[line:   7, col:  60] -> typeid: 2	type: [id]	info: "list"
[line:   7, col:  61] -> typeid: 98	symbol: ","
[line:   7, col:  62] -> typeid: 2	type: [id]	info: "i"
[line:   7, col:  63] -> typeid: 14	symbol: ")"
[line:   7, col:  64] -> typeid: 14	symbol: ")"
[line:   7, col:  65] -> typeid: 101	symbol: ";"
[line:   8, col:   1] -> typeid: 102	symbol: "[newline]"

[line:   8, col:   2] -> typeid: 18	symbol: "}"
[line:   9, col:   1] -> typeid: 102	symbol: "[newline]"

[line:   9, col:   5] -> typeid: 2	type: [id]	info: "puts"
[line:   9, col:   6] -> typeid: 13	symbol: "("
[line:   9, col:   8] -> typeid: 10	type: [string]	info: """"
[line:   9, col:   9] -> typeid: 14	symbol: ")"
[line:   9, col:  10] -> typeid: 101	symbol: ";"
[line:  10, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  10, col:   2] -> typeid: 18	symbol: "}"
[line:  11, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  11, col:   4] -> typeid: 35	symbol: "int"
[line:  11, col:   8] -> typeid: 2	type: [id]	info: "main"
[line:  11, col:   9] -> typeid: 13	symbol: "("
[line:  11, col:  10] -> typeid: 14	symbol: ")"
[line:  12, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  12, col:   2] -> typeid: 17	symbol: "{"
[line:  13, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  13, col:  10] -> typeid: 2	type: [id]	info: "llist_int"
[line:  13, col:  11] -> typeid: 79	symbol: "*"
[line:  13, col:  15] -> typeid: 2	type: [id]	info: "list"
[line:  13, col:  16] -> typeid: 63	symbol: "="
[line:  13, col:  31] -> typeid: 2	type: [id]	info: "create_list_int"
[line:  13, col:  32] -> typeid: 13	symbol: "("
[line:  13, col:  33] -> typeid: 3	symbol: "0"
[line:  13, col:  34] -> typeid: 14	symbol: ")"
[line:  13, col:  35] -> typeid: 101	symbol: ";"
[line:  14, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  14, col:   4] -> typeid: 32	symbol: "for"
[line:  14, col:   5] -> typeid: 13	symbol: "("
[line:  14, col:   8] -> typeid: 35	symbol: "int"
[line:  14, col:   9] -> typeid: 2	type: [id]	info: "i"
[line:  14, col:  10] -> typeid: 63	symbol: "="
[line:  14, col:  11] -> typeid: 4	type: [dec]	info: "1"
[line:  14, col:  12] -> typeid: 101	symbol: ";"
[line:  14, col:  13] -> typeid: 2	type: [id]	info: "i"
[line:  14, col:  14] -> typeid: 92	symbol: "<"
[line:  14, col:  16] -> typeid: 4	type: [dec]	info: "10"
[line:  14, col:  17] -> typeid: 101	symbol: ";"
[line:  14, col:  19] -> typeid: 74	symbol: "++"
[line:  14, col:  20] -> typeid: 2	type: [id]	info: "i"
[line:  14, col:  21] -> typeid: 14	symbol: ")"
[line:  15, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  15, col:   2] -> typeid: 17	symbol: "{"
[line:  16, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  16, col:  19] -> typeid: 2	type: [id]	info: "push_back_list_int"
[line:  16, col:  20] -> typeid: 13	symbol: "("
[line:  16, col:  24] -> typeid: 2	type: [id]	info: "list"
[line:  16, col:  25] -> typeid: 98	symbol: ","
[line:  16, col:  26] -> typeid: 2	type: [id]	info: "i"
[line:  16, col:  27] -> typeid: 14	symbol: ")"
[line:  16, col:  28] -> typeid: 101	symbol: ";"
[line:  17, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  17, col:   2] -> typeid: 18	symbol: "}"
[line:  18, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  18, col:   5] -> typeid: 2	type: [id]	info: "pout"
[line:  18, col:   6] -> typeid: 13	symbol: "("
[line:  18, col:  10] -> typeid: 2	type: [id]	info: "list"
[line:  18, col:  11] -> typeid: 14	symbol: ")"
[line:  18, col:  12] -> typeid: 101	symbol: ";"
[line:  19, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  19, col:  22] -> typeid: 2	type: [id]	info: "insert_index_list_int"
[line:  19, col:  23] -> typeid: 13	symbol: "("
[line:  19, col:  27] -> typeid: 2	type: [id]	info: "list"
[line:  19, col:  28] -> typeid: 98	symbol: ","
[line:  19, col:  29] -> typeid: 4	type: [dec]	info: "5"
[line:  19, col:  30] -> typeid: 98	symbol: ","
[line:  19, col:  33] -> typeid: 4	type: [dec]	info: "999"
[line:  19, col:  34] -> typeid: 14	symbol: ")"
[line:  19, col:  35] -> typeid: 101	symbol: ";"
[line:  20, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  20, col:  22] -> typeid: 2	type: [id]	info: "insert_index_list_int"
[line:  20, col:  23] -> typeid: 13	symbol: "("
[line:  20, col:  27] -> typeid: 2	type: [id]	info: "list"
[line:  20, col:  28] -> typeid: 98	symbol: ","
[line:  20, col:  29] -> typeid: 4	type: [dec]	info: "3"
[line:  20, col:  30] -> typeid: 98	symbol: ","
[line:  20, col:  33] -> typeid: 4	type: [dec]	info: "777"
[line:  20, col:  34] -> typeid: 14	symbol: ")"
[line:  20, col:  35] -> typeid: 101	symbol: ";"
[line:  21, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  21, col:   5] -> typeid: 2	type: [id]	info: "pout"
[line:  21, col:   6] -> typeid: 13	symbol: "("
[line:  21, col:  10] -> typeid: 2	type: [id]	info: "list"
[line:  21, col:  11] -> typeid: 14	symbol: ")"
[line:  21, col:  12] -> typeid: 101	symbol: ";"
[line:  22, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  23, col:   5] -> typeid: 2	type: [id]	info: "delete_list_node_int"
[line:  23, col:   6] -> typeid: 13	symbol: "("
[line:  23, col:  10] -> typeid: 2	type: [id]	info: "list"
[line:  23, col:  11] -> typeid: 98	symbol: ","
[line:  23, col:  12] -> typeid: 4	type: [dec]	info: "8"
[line:  23, col:  13] -> typeid: 14	symbol: ")"
[line:  23, col:  14] -> typeid: 101	symbol: ";"
[line:  24, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  24, col:  21] -> typeid: 2	type: [id]	info: "delete_list_node_int"
[line:  24, col:  22] -> typeid: 13	symbol: "("
[line:  24, col:  26] -> typeid: 2	type: [id]	info: "list"
[line:  24, col:  27] -> typeid: 98	symbol: ","
[line:  24, col:  28] -> typeid: 4	type: [dec]	info: "8"
[line:  24, col:  29] -> typeid: 14	symbol: ")"
[line:  24, col:  30] -> typeid: 101	symbol: ";"
[line:  25, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  25, col:   5] -> typeid: 2	type: [id]	info: "pout"
[line:  25, col:   6] -> typeid: 13	symbol: "("
[line:  25, col:  10] -> typeid: 2	type: [id]	info: "list"
[line:  25, col:  11] -> typeid: 14	symbol: ")"
[line:  25, col:  12] -> typeid: 101	symbol: ";"
[line:  26, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  26, col:  16] -> typeid: 2	type: [id]	info: "delete_list_int"
[line:  26, col:  17] -> typeid: 13	symbol: "("
[line:  26, col:  21] -> typeid: 2	type: [id]	info: "list"
[line:  26, col:  22] -> typeid: 14	symbol: ")"
[line:  26, col:  23] -> typeid: 101	symbol: ";"
[line:  27, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  27, col:   7] -> typeid: 38	symbol: "return"
[line:  27, col:   8] -> typeid: 3	symbol: "0"
[line:  27, col:   9] -> typeid: 101	symbol: ";"
[line:  28, col:   1] -> typeid: 102	symbol: "[newline]"

[line:  28, col:   2] -> typeid: 18	symbol: "}"
